target_chains/ethereum/contracts/contracts/pulse/Pulse.sol

```diff
@@ -105,15 +124,6 @@ abstract contract Pulse is IPulse, PulseState {
 
         clearRequest(sequenceNumber);
 
-        // Check if enough gas remains for callback + events/cleanup
-        // We need extra gas beyond callbackGasLimit for:
-        // 1. Emitting success/failure events
-        // 2. Error handling in catch blocks
-        // 3. State cleanup operations
-        if (gasleft() < (req.callbackGasLimit * 3) / 2) {
-            revert InsufficientGas();
-        }
-
```

Comment:
removed this because if insufficient gas error occurs within callback then `PriceUpdateCallbackFailed` will be emitted with `"low-level error (possibly out of gas)"` and otherwise if insufficient gas for the `executeCallback` function itself, it will revert with out-of-gas